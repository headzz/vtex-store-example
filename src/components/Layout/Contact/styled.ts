import { colors } from 'configuration'
import styled from 'styled-components'

export const ContactStyled = styled.section`
  background-color: ${colors.PINK_JUSSI};
  padding: 13.6rem 0;
  width: 100%;
`
