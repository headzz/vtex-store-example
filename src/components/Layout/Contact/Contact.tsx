import { ContactStyled } from './styled'

/**
 * @component
 * @name Contact
 * @description Defines the container for the layout of contact section.
 * @namespace components components/Layout/Contact
 * @param {element} children Component(s) that may appear inside the layout.
 *
 */

const Contact: React.FC = ({ children }) => {
  return <ContactStyled>{children}</ContactStyled>
}

export default Contact
