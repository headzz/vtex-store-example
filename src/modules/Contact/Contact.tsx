import Layout from 'components/Layout'

import TextArea from './components/TextArea'
import { ContactContainer } from './styled'

/**
 * @component
 * @name Contact
 * @description creates Contact component section to use on screen.
 * @namespace components modules/Contact
 *
 */
const Contact: React.FC = () => (
  <Layout.Contact>
    <ContactContainer>
      <TextArea />
    </ContactContainer>
  </Layout.Contact>
)

export default Contact
