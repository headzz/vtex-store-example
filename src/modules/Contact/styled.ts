import styled from 'styled-components'

export const ContactContainer = styled.div`
  margin: 0 auto;
  div {
    padding: 0 1rem;
    width: 100%;

    p:first-child {
      margin-bottom: 3.2rem;
      text-align: center;
    }

    p:nth-child(2) {
      margin-bottom: 0.8rem;
    }
  }

  @media (min-width: 768px) {
    div {
      padding: 0;
      width: 67.6rem;
    }
  }
`
