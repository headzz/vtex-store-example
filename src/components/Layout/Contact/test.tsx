import { render } from '@testing-library/react'

import Contact from './Contact'

describe('<Contact />', () => {
  it('should render component that wraps Contact on page', () => {
    const { container } = render(<Contact />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
