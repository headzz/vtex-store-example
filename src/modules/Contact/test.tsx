import { render } from '@testing-library/react'

import Contact from './Contact'

describe('Contact', () => {
  it('match snapshot', () => {
    const { container } = render(<Contact />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
