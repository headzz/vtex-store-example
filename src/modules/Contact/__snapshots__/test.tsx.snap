// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact match snapshot 1`] = `
.c0 {
  background-color: #EE0E7D;
  padding: 13.6rem 0;
  width: 100%;
}

.c2 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  margin: inherit;
  width: 100%;
}

.c3 {
  color: #ffffff;
  font-size: 3.2rem;
  font-weight: 700;
}

.c4 {
  color: #ffffff;
  font-size: 1.6rem;
  font-weight: 500;
}

.c5 {
  color: #ffffff;
  font-size: 3.2rem;
  font-weight: 500;
}

.c1 {
  margin: 0 auto;
}

.c1 div {
  padding: 0 1rem;
  width: 100%;
}

.c1 div p:first-child {
  margin-bottom: 3.2rem;
  text-align: center;
}

.c1 div p:nth-child(2) {
  margin-bottom: 0.8rem;
}

@media (min-width:768px) {
  .c1 div {
    padding: 0;
    width: 67.6rem;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <p
        class="c3"
        color="#ffffff"
        title="small"
      >
        ESSA LOJA FOI CONSTRUÍDA USANDO UMA DAS NOSSAS SOLUÇÕES DA PLATAFORMA VTEX. TENHA A SUA.
      </p>
      <p
        class="c4"
        color="#ffffff"
        title=""
      >
        Entre em contato
      </p>
      <p
        class="c5"
        color="#ffffff"
        title="small"
      >
        comercial@jussi.com.br
      </p>
    </div>
  </div>
</section>
`
